src/AddItems.js

```diff
@@ -1,25 +1,46 @@
 import React, { useState } from 'react';
 import { db } from './lib/firebase';
+import { useSnackbar } from 'notistack';
 
 function AddItems(props) {
   const [groceryItem, setGroceryItem] = useState('');
-  const userToken = localStorage.getItem('userToken');
   const [itemFreq, setItemFreq] = useState(7);
 
+  const { enqueueSnackbar } = useSnackbar();
+
   const updateGroceryItem = (event) => {
     setGroceryItem(event.target.value);
   };
 
+  const checkItem = (item) => {
+    const itemLetters = item.toLowerCase().split('');
+    const itemLettersFilt = itemLetters.filter((l) => {
+      return /[a-z|\s]/.test(l);
+    });
+    return itemLettersFilt.join('');
+  };
+
   const submitGroceryItem = (event) => {
     event.preventDefault();
     const formData = {
       itemName: groceryItem,
       frequency: itemFreq,
       lastPurchaseDate: null,
     };
-    db.collection(userToken).add({ formData });
-    setGroceryItem('');
-    setItemFreq(7);
+
+    const filtered = props.list.filter((existingItem) => {
+      return checkItem(existingItem) === checkItem(groceryItem);
+    });
+
+    if (filtered.length > 0) {
+      enqueueSnackbar('This item is already on your list', {
+        variant: 'error',
+      });
+    } else {
+      db.collection(props.userToken).add({ formData });
+      setGroceryItem('');
+      setItemFreq(7);
```

Comment:
I like the fact that you passed the list as props to the `AddItems` component. Out of curiosity, is there any benefit to moving this to the `App` component so that all firebase calls are being done there?

Reply:
I think what we are starting to see is that the `ItemList` component requires the list from the Firestore for displaying the data, and the `AddItems` component also requires that same list for validating entries before submitting the form. So instead of calling the Firestore for the same list every time we navigate pages, we can reduce the number of calls by saving that data in a parent component. Great optimizations here!

Reply:
Yup, that's exactly why we pulled it up to `App` - we realized we were going to be calling to Firestore in both `ItemList` and `AddItems`